### PR TITLE
WIP: Create Power-specific Node class.

### PR DESCRIPTION
**Pull request by PeXXeR**
isRotateAndMask(TR::Node* node) is a function present inside common
OMR::CodeGenerator. This is a Power architecture question and pattern
match only, and really should be asked on the TR::Node itself.
Hence, need to relocate this API to power specific node class.
Since, making this complete change will require change in p.mk file in
OpenJ9 as well.

So, as part of this commit, just publishing the skeleton of OMRNode Class.
Once this gets in, will make PR for OpenJ9 and get back to final changes
with isRotateAndMask().

Issue: #1876

Signed-off-by: Somesh Sharma <someshsharma0312@gmail.com>

### CODE REVIEW

**Comment by PeXXeR** [comment]
Investigating failures.

**Comment by PeXXeR** [comment]
> Please use the imperative mood, consistent capitalization, and remove the ending period from the commit title (e.g. `Create Power-specific Node class`).
> 
> Also, while I understand this change is needed in order to move `isRotateAndMask()`, I don't think it should be mentioned in the commit message for this change.

I have made the requested changes for title and description of the commit : 1fe1153

File: compiler/p/il/OMRNode.hpp [hunk 1]
```diff
@@ -0,0 +1,66 @@
+/*******************************************************************************
+ * Copyright (c) 2000, 2019 IBM Corp. and others
```
[the remark this answers is not in the document]
Comment (PeXXeR) [reply compiler/p/il/OMRNode.hpp]: Addressed in 0e8bf58

File: compiler/p/il/OMRNode.cpp [hunk 2]
```diff
@@ -0,0 +1,23 @@
+/*******************************************************************************
+ * Copyright (c) 2000, 2019 IBM Corp. and others
```
[the remark this answers is not in the document]
Comment (PeXXeR) [reply compiler/p/il/OMRNode.cpp]: Addressed in ab7f1cd

File: compiler/p/il/OMRNode.hpp [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+{
+
+namespace Power
+{
+
+class OMR_EXTENSIBLE Node : public OMR::Node
+   {
+protected:
+
+   Node() : OMR::Node() {}
```
[the remark this answers is not in the document]
Comment (PeXXeR) [reply compiler/p/il/OMRNode.hpp]: Addressed in ab7f1cd

**Comment by andrewcraik** [comment]
Per my comments on the original issue - I don't think creating a POWER specific subclass of Node is the right thing to do to accommodate this function. Firstly, it is not a property of a node but a pattern match of nodes which is unlike most Node APIs. Secondly, the notion could be useful to the optimizer and casing the IL per codegen will increase fragmentation in the codebase and make it harder for people to contribute implementations to multiple code generators when adding new capabilities. Finally, the current implementation references a number of deprecated opcodes that I don't think should be enshrined into a refactored representation. The intention of moving this query to somewhere better is good, but I don't think this is the right approach.

**Comment by PeXXeR** [comment]
Closing this PR as these changes are not relevant.
Will open a new one without power specific node class.

**Pull request closed without merging**